packaging/userspace/libdirectives.py

```diff
@@ -205,24 +213,42 @@ def read_directives_for_target(self, target_name: str) -> NoReturn:
 						soname = directive_l[1:]
 						self.allowlist.sonames.discard(soname)
 						self.denylist.sonames.add(soname)
+					elif directive_l[0] == '~':
+						soname = directive_l[1:]
+						self.allowlist.sonames.discard(soname)
+						self.allowlist.sonames.discard(soname)
 					else:
 						l.warning('%s:%s: Syntax error, ignoring line', soname_dfname, str(lnum))
-		else:
-			l.warning('No soname directives file for %s target_name. '
-			          'This will likely result in the exclusion of required libraries.',
-			          target_name)
+			return True
+		return False
 
 	def read_directives(self) -> NoReturn:
 		l = logging.getLogger(__name__)  # noqa: VNE001,E741
 		l.info('reading known libs data')
 
 		self.read_directives_for_target(target_name='common')
-		self.read_directives_for_target(target_name=self.target_platform)
+
+		# FIXME: Account for populated target_platform_base but empty target_platform
```

Comment:
still need to be handled?

Reply:
Doesn't come up with the current directive files.